From bd2c5ec61f596bd564d0b5c498b6d44962fb8b12 Mon Sep 17 00:00:00 2001
From: wenbin <wenbin_mj@163.com>
Date: Thu, 10 Sep 2026 21:16:40 +0800
Subject: [PATCH 3/3] ohos: provide APP_HOME_DIR on HarmonyOS

APP_HOME_DIR was defined only for android/ios. HarmonyOS sets it through the same
mobile path (flutter_ffi's main_set_home_dir), so it needs to exist there too.
---
 src/config.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/config.rs b/src/config.rs
index c3e85e8..a8f70ba 100644
--- a/src/config.rs
+++ b/src/config.rs
@@ -92,7 +92,7 @@ lazy_static::lazy_static! {
     pub static ref APP_DIR: RwLock<String> = Default::default();
 }
 
-#[cfg(any(target_os = "android", target_os = "ios"))]
+#[cfg(any(target_os = "android", target_os = "ios", target_env = "ohos"))]
 lazy_static::lazy_static! {
     pub static ref APP_HOME_DIR: RwLock<String> = Default::default();
 }
